### PR TITLE
docs: update ts docs field to `@func`

### PR DESCRIPTION
**Pull request by TomChv**
Follows up #7701 with a doc update

### CODE REVIEW

File: docs/current_docs/manuals/developer/snippets/state-functions/typescript/index.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
-import { dag, Container, object, func, field } from "@dagger.io/dagger"
+import { object, func } from "@dagger.io/dagger"
```
Comment (gerhard) [inline docs/current_docs/manuals/developer/snippets/state-functions/typescript/index.ts:1]: Was `dag` & `Container` redundant here?

**Review by gerhard — APPROVED** [review]
Apart from that one comment, everything else looks good to me. WDYT @helderco ?

File: docs/current_docs/manuals/developer/snippets/state-functions/typescript/index.ts (hunk 1 above)
Comment (TomChv) [reply docs/current_docs/manuals/developer/snippets/state-functions/typescript/index.ts:1]: It was not used.

**Comment by helderco** [comment]
This should only be merged after https://github.com/dagger/dagger/pull/7701 is released. By that time there could be more doc snippets to update. That's why I'm holding off review until then.

**Comment by jedevc** [comment]
@helderco should we take this and re-deploy docs?

**Pull request merged**